notebooks/nessie-iceberg-demo-nba.ipynb

```diff
@@ -0,0 +1,629 @@
+    "    spark.sql(\"SHOW TABLES IN dev_catalog\").toPandas()\n",
+    "except:\n",
+    "    print(\"Error in 0.9.0 of Nessie causes an exception when no tables found\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And on the `dev` branch we expect to see two tables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"USE REFERENCE dev IN dev_catalog\").toPandas()\n",
+    "spark.sql(\"SHOW TABLES IN dev_catalog\").toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "We can also verify that the `dev` and `main` branches point to different commits"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"LIST REFERENCES IN dev_catalog\").toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Dev promotion into main\n",
+    "-----------------------\n",
+    "Once we are done with our changes on the `dev` branch, we would like to merge those changes into `main`.\n",
+    "We merge `dev` into `main` via the Spark sql `merge` command.\n",
+    "Both branches should be at the same revision after merging/promotion."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"MERGE BRANCH dev IN dev_catalog\").toPandas()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can verify the branches are at the same hash and that the `main` branch now contains the expected tables and row counts.\n",
+    "\n",
+    "The tables are now on `main` and ready for consumtion by our blog authors and analysts!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"LIST REFERENCES IN dev_catalog\").toPandas()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spark.sql(\"USE REFERENCE main IN dev_catalog\").toPandas()\n",
+    "spark.sql(\"SHOW TABLES IN dev_catalog\").toPandas()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.salaries\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from salaries_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count\n",
+    "\n",
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.totals_stats\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from stats_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count"
```

Comment:
Same here: Maybe add some print of these numbers here, so people have something to look at.

Reply:
good idea, done